### ISSUE TITLE
move growth_rate_given_height to Plant/Strategy classes

### ISSUE THREAD

**Issue by rafaqz** [traitecoevo/plant#222]
https://github.com/traitecoevo/plant/blob/ee692a705e3c444207c361ee1d6424ee5aca8e98/inst/include/plant/cohort.h#L211

There is some code in `Cohort` that really should be in `Plant`. `growth_rate_given_height`
is all `plant.xxx` methods already. 

It also seems to be the `size_dt` method we were talking about, if I get what's happening here.

**Comment by dfalster** [comment]
Hi, Just been reviewing this. You're right, this function does give growth rate, but you shouldn't move it. The purpose of having it here is that growth rate is used elsewhere in some root-solving routines, which happen at the plant level and for which we need to call growth rate. Also worth noting, you can't calculate growth rate without first having a state -- in this case height. 

If you look in strategy, all 3 of the vital rates are returned in the`compute _rates` function. Growth rate is calculated at following point, with Mortality and fecundity on lines just before/after:

https://github.com/rafaqz/plant/blob/52f23cfbfb7740f866c7748185588151943ffefe/src/ff16_strategy.cpp#L185

The reason we put all three in this function is so we could reuse bits of the calculated variables, in particular `net_mass_production_dt_` which is expensive to calculate and influences all three. 

So perhaps the best advice for the generic strategy should be to have a function `compute_rates` that computes all three and fills the `vars` vector with these values.   

Should re rename `vars` too? Perhaps to `rates`?  It's owned by the `plant` object, defined here

https://github.com/rafaqz/plant/blob/abstract_strategies3/inst/include/plant/plant.h#L117

And as discussed, we could rename the `plant` layer to `individual`. (this may require quite a few changes in tests and docs)

**Comment by rafaqz** [comment]
Oh I meant it could be a method of Plant (so with state), that maybe called a Strategy method. It just seems strange on Cohort. It's often considered a code smell when a method only calls methods on another class and doesn't use its own member variables at all - a method that only accesses Plant member variables would normally be a method of Plant.

So instead of calling:
```c
growth_rate_given_height(p, h, environment);
```
It would be:
```c
p.growth_rate_given_height(h, environment);
```
Unless I misunderstand where state is.

I agree renaming `vars` would be a nice reduction in the namespace. And yes plant -> individual. germination -> establishment was also discussed on saturday.

**Comment by dfalster** [comment]
Ah, I misread, thought you were moving to strategy.

Sure, if that works & is improvement.

In any case, now we know why there is no explicit function in strategy for growth rate.

--
Daniel Falster

Deputy Director  (Graduate Program) Evolution & Ecology Research Centre | ARC Future Fellow
School of Biological, Earth & Environmental Sciences | University of New South Wales
P:+61-2-9385 8431 | A: Bdlg E26 Rm 5109<https://goo.gl/maps/iS3Jc2YmYu62>  | T: @adaptive_plant<https://twitter.com/adaptive_plant> | W: http://danielfalster.com<http://danielfalster.com/>

**Comment by dfalster** [comment]
Thinking more about the `vars` variables, it's not exclusively `rates` so hence why we called it `vars`, short for `variables`.

**Comment by dfalster** [comment]
has been implemented in @rafaqz's fork, to be merged in near future.

**Comment by dfalster** [comment]
Addressed via #235
